File: backend/src/main/java/br/com/easytask/Easy/Task/repository/TaskRepository.java
package br.com.easytask.Easy.Task.repository;

import br.com.easytask.Easy.Task.model.Task;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface TaskRepository extends JpaRepository<Task, Long> {

}

File: app/src/main/java/br/com/fiap/easytask/repository/TaskRepository.kt
package br.com.fiap.easytask.repository

import br.com.fiap.easytask.model.Task
import br.com.fiap.easytask.network.TaskApiService

class TaskRepository {
    private val apiService = TaskApiService.api

    fun getTasks(): retrofit2.Call<List<Task>> = apiService.getTasks()

    fun createTask(task: Task): retrofit2.Call<Task> = apiService.createTask(task)

    fun getTaskById(id: Long): retrofit2.Call<Task> = apiService.getTaskById(id)

    fun updateTask(id: Long, task: Task): retrofit2.Call<Task> = apiService.updateTask(id, task)

    fun deleteTask(id: Long): retrofit2.Call<Void> = apiService.deleteTask(id)
}
